From 23eca5b8103b4c6c74ce5a4c9a73180ded5f33ca Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Matthew=20=F0=9F=AA=B7?= <matthew@urgero.net>
Date: Sat, 14 Dec 2024 15:23:58 -0600
Subject: [PATCH 06/46] Update Ruth Epoch IFP address (#7)

Previous v8.3.3 implementation used the same IFP address from the Judges
Epoch, but a new address is necessary in order to determine the fork
block into the Ruth Epoch. This commit adds a new IFP address for this
purpose.
---
 src/consensus/addresses.h                             | 2 +-
 test/functional/logos_feature_minerfund_activation.py | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/consensus/addresses.h b/src/consensus/addresses.h
index cfa91d497..59c539051 100644
--- a/src/consensus/addresses.h
+++ b/src/consensus/addresses.h
@@ -143,7 +143,7 @@ Consensus::CoinbaseAddresses AddressSets = {
         },
     .ruth =
         {
-            "lotus_16PSJMaps9sQg7aBQgyY1RdHb2fZYdmWhQPbgus75"
+            "lotus_16PSJPi88MtH34Ti3dZza4MFRF9XUVd3fKc6Ec3TV"
         }
 };
 } // namespace RewardAddresses
diff --git a/test/functional/logos_feature_minerfund_activation.py b/test/functional/logos_feature_minerfund_activation.py
index 967b58fd7..2dfb7e141 100755
--- a/test/functional/logos_feature_minerfund_activation.py
+++ b/test/functional/logos_feature_minerfund_activation.py
@@ -128,7 +128,7 @@ JUDGES_SCRIPTS = [
     "76a9149208ecc785c92968481a92aee7024b77e54d27dc88ac",
 ]
 RUTH_SCRIPTS = [
-    "76a9149208ecc785c92968481a92aee7024b77e54d27dc88ac",
+    "76a914d7756030f4292b70cb70dc4df231b14109c6806188ac",
 ]
 
 class MinerFundActivationTest(BitcoinTestFramework):
-- 
2.39.5 (Apple Git-154)

